src/App.tsx

```diff
@@ -26,70 +27,104 @@ type ReorderOptions = {
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  } else if (sortType === SortType.ALPABET) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: 0,
+  };
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+  render(): React.ReactNode {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': this.state.sortType !== SortType.ALPABET,
+            })}
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPABET });
+            }}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': this.state.sortType !== SortType.LENGTH,
+            })}
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH });
+            }}
+          >
+            Sort by length
+          </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': !this.state.isReversed,
+            })}
+            onClick={() => {
+              this.setState((state) => ({
+                isReversed: !state.isReversed,
+              }));
+            }}
+          >
+            Reverse
+          </button>
+
+          {(this.state.sortType !== SortType.NONE || this.state.isReversed) && (
```

Comment:
create separate variable for condition. You should always think of what happens if we need to add some new condition here. You will have to look for this piece of code manually. Better to keep it in one place